tortoise/verifying_algorithm.go

```diff
@@ -0,0 +1,88 @@
+package tortoise
+
+import (
+	"sync"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/mesh"
+)
+
+// ThreadSafeVerifyingTortoise is a thread safe verifying tortoise wrapper, it just locks all actions.
+type ThreadSafeVerifyingTortoise struct {
+	trtl  *turtle
+	mutex sync.Mutex
+}
+
+// NewVerifyingTortoise creates a new verifying tortoise wrapper
+func NewVerifyingTortoise(layerSize int, mdb blockDataProvider, hdist int, lg log.Log) *ThreadSafeVerifyingTortoise {
```

Comment:
consider consolidating two functions

Reply:
which two? this and the recovered one? recovered one loads hdist and layersize from disc. we can consolidate and add some boolean to distinguish but hdist and layersize will stay unused in the case of a recovered tortoise.

Reply:
yes, I think we should consolidate both